test/dataframe.jl

```diff
@@ -1127,6 +1127,16 @@ end
     @test names(df) == [:A_4, :B_4]
     @test rename!(x->Symbol(lowercase(string(x))), df) === df
     @test names(df) == [:a_4, :b_4]
+
+    df = DataFrame(A = 1:3, B = 'A':'C', C = [:x, :y, :z])
+    @test rename!(df, :A => :B, :B => :A) === df
+    @test names(df) == [:B, :A, :C]
+    @test rename!(df, :A => :B, :B => :A, :C => :D) === df
+    @test names(df) == [:A, :B, :D]
+    @test rename!(df, :A => :B, :B => :C, :D => :A) === df
+    @test names(df) == [:B, :C, :A]
+    @test rename!(df, :A => :B, :B => :C, :C => :A) === df
+    @test names(df) == [:C, :A, :B]
```

Comment:
Can you please also add tests for each case of the error.
Ideally also randomized tests could be added here to make sure we cover all corner cases as the logic is not simple.